modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteProject.java

```diff
@@ -123,7 +125,7 @@ public Project copy(RelTraitSet traitSet, RelNode input, List<RexNode> projects,
         }
 
         if (srcKeys.size() == keys.size()) {
-            return Pair.of(nodeTraits, List.of(in.replace(hash(srcKeys, distribution.function()))));
+            return Pair.of(nodeTraits, List.of(in.replace(hash(ImmutableIntList.copyOf(srcKeys), distribution.function()))));
```

Comment:
do we need a copy here?
```suggestion
            return Pair.of(nodeTraits, List.of(in.replace(hash(ImmutableIntList.of(srcKeys.elements()), distribution.function()))));
```